src/components/PasteButton.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { GestureResponderEvent, StyleSheet, Text } from 'react-native'
+import Touchable from 'src/components/Touchable'
+import { Colors } from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { useClipboard } from 'src/utils/useClipboard'
+
+interface Props {
+  value: string
+  setValue: React.Dispatch<React.SetStateAction<string>>
+  onPress?: (event: GestureResponderEvent) => void
+}
+
+export const PasteButton = ({ value, setValue, onPress }: Props) => {
```

Comment:
i'm consumed about these props, it sounds like a consumer would do something like `const [value, setValue] = useState(something)` and pass the `value` and `setValue` in to here? why? passing state setters directly into a component is usually a sign that the abstraction doesn't make sense. i'd expect a paste button to simply paste what's in the clipboard and it should not worry about any values.

i'd recommend that this takes 1 prop (onPress) and the consumer should do the `setValue` inside the onPress.

Reply:
That makes sense. I was considering removing those properties as well. I've made the change.

I've manually tested in SupportContact with the below code. It works.
```ts
            <TextInput
              onChangeText={setName}
              multiline={false}
              value={name}
              style={styles.singleLineTextInput}
              showClearButton={true}
              testID={'NameEntry'}
              rightElement={ !name &&
                <PasteButton onPress={(content, event) => {
                  setName(content)
                  ValoraAnalytics.track(SettingsEvents.settings_revoke_phone_number_confirm)
                }} />
              }
            />
```